fearless/Modules/Staking/SelectValidatorsFlow/SelectedValidatorList/SelectedValidatorListViewController.swift

```diff
@@ -0,0 +1,240 @@
+
+    private func setupProceedButton() {
+        rootView.proceedButton.addTarget(self, action: #selector(tapProceedButton), for: .touchUpInside)
+        updateProceedButton()
+    }
+
+    private func updateEditButton() {
+        if rootView.tableView.isEditing {
+            navigationItem.rightBarButtonItem?.title = R.string.localizable
+                .commonDone(preferredLanguages: selectedLocale.rLanguages)
+        } else {
+            navigationItem.rightBarButtonItem?.title = R.string.localizable
+                .commonEdit(preferredLanguages: selectedLocale.rLanguages)
+        }
+    }
+
+    private func updateProceedButton() {
+        let cellViewModels = viewModel?.cellViewModels ?? []
+
+        let buttonTitle: String
+        let enabled: Bool
+        let fillColor: UIColor
+
+        if cellViewModels.count > selectedValidatorsLimit {
+            enabled = false
+            fillColor = R.color.colorDarkGray()!
+            buttonTitle = R.string.localizable
+                .stakingCustomProceedButtonDisabledTitle(
+                    selectedValidatorsLimit,
+                    preferredLanguages: selectedLocale.rLanguages
+                )
+
+        } else {
+            enabled = true
+            fillColor = R.color.colorAccent()!
+            buttonTitle = R.string.localizable
+                .commonContinue(
+                    preferredLanguages: selectedLocale.rLanguages
+                )
+        }
+
+        rootView.proceedButton.triangularedView?.fillColor = fillColor
+        rootView.proceedButton.imageWithTitleView?.title = buttonTitle
+        rootView.proceedButton.isEnabled = enabled
+    }
+
+    private func updateHeaderView() {
+        guard let viewModel = viewModel else { return }
+        headerView?.bind(
+            viewModel: viewModel.headerViewModel,
+            shouldAlert: viewModel.limitIsExceeded
+        )
+    }
+
+    private func presentValidatorInfo(at index: Int) {
+        presenter.didSelectValidator(at: index)
+    }
+
+    // MARK: - Actions
+
+    @objc private func tapEditButton() {
+        rootView.tableView.setEditing(
+            !rootView.tableView.isEditing,
+            animated: true
+        )
+        updateEditButton()
+        updateProceedButton()
+    }
+
+    @objc private func tapProceedButton() {
+        presenter.proceed()
+    }
+}
+
+// MARK: - Localizable
+
+extension SelectedValidatorListViewController: Localizable {
+    func applyLocalization() {
+        if isViewLoaded {
+            title = R.string.localizable
+                .stakingSelectedValidatorsTitle(preferredLanguages: selectedLocale.rLanguages)
+
+            updateEditButton()
+            updateProceedButton()
+            updateHeaderView()
+        }
+    }
+}
+
+// MARK: - SelectedValidatorListViewProtocol
+
+extension SelectedValidatorListViewController: SelectedValidatorListViewProtocol {
+    func updateViewModel(_ viewModel: SelectedValidatorListViewModel) {
```

Comment:
I didn't saw that it is called at some point when reload is not called. So, can we combine them?

Reply:
It's called whenever we remove a row. In this case we don't want to reload tableView since it will cause data inconsistency